### DISCUSSION TITLE
Configure some ruleset

### DISCUSSION THREAD

**Discussion by RSickenberg** [VincentLanglet/Twig-CS-Fixer#52] [Ideas]
I'd like to change the spaces between some operators like `{}`. Is it something we could add?

**Comment by VincentLanglet** [comment]
This library was made to follow the twig standard https://twig.symfony.com/doc/3.x/coding_standards.html
which give this example
<img width="214" alt="image" src="https://user-images.githubusercontent.com/9052536/181737844-840bcf3d-7184-45c0-a818-f899d33a82ac.png">
I'm not sure it's a good idea to promote using another config, to avoid https://xkcd.com/927/

But I wont block a PR. If you want to change space around `{}`, you'll need to add config in the constructor of PunctuationSpacingSniff
https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/src/Sniff/PunctuationSpacingSniff.php.

The main difficulty will be to find a nice way to represent
```
    private const NO_SPACE_BEFORE = [')', ']', '}', ':', '.', ',', '|'];
    private const NO_SPACE_AFTER = ['(', '[', '{', '.', '|'];
    private const ONE_SPACE_AFTER = [':', ','];
```
as a config.

**Reply by RSickenberg** [reply]
Yeah, right I agree. (Nice comic btw, forgot about xkcd) 😁
Let's keep it simple and follow the standards.

Many thanks for this tool btw, we stick too long to https://github.com/trivago/prettier-plugin-twig-melody, and now finding a PHP tool is incredible.

Let me know If you need help contributing, it's the minimum I can do.

**Reply by VincentLanglet** [reply]
> Let me know If you need help contributing, it's the minimum I can do.

Thanks. The projet didn't change a lot recently because I got no new need or bugs.
But I'm always open to suggestions/issues/PR.

If you like this repo feel free to star/sponsor/share :)

**Reply by RSickenberg** [reply]
As long as the project is maintained it's okay for me too :)

**Comment by RSickenberg** [comment]
Are you ok to see if it's possible to sync with https://github.com/friendsoftwig/twigcs rulesets?
When I fix there is still errors/warning remaining mostly due to spaces before `with` or after the `=` and mostly due to prettier-melody that thought it was a good idea to shift everything to a new line.

**Reply by VincentLanglet** [reply]
I'll need example of issue you get

**Reply by RSickenberg** [reply]
Here an example: https://github.com/friendsoftwig/twigcs/issues/254

My bet, the issue is on twigcs hand but the fix can be on yours.
For now, I can live with this :-)

**Reply by VincentLanglet** [reply]
I still don't understand what you're expecting for my tool.

Can you give me the code before running my tool.
And the code you expect to get after running my tool.

Like:
I expect
```
{% embed 'components/table.html.twig'
with {} %}
```
to be changed to
```
{% embed 'components/table.html.twig' with {} %}
```

**Reply by RSickenberg** [reply]
In fact, it's exactly what I wanted 😊 (from your example)

**Reply by VincentLanglet** [reply]
I think that, for all my rules, I ignore the new lines, because people might prefer multi line, shorter line, etc.
Cf check `$this->isTokenMatching($tokens[$next], Token::EOL_TOKENS)`.
https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/src/Sniff/AbstractSpacingSniff.php#L72

If it's really wanted, it would need to add config in order to ignore or not new lines, 
but it would have too much impact IMHO.

Like you said, I think the issue is more about fos/twigcs which doesn't support new lines.
